### PR TITLE
:bug: :white_check_mark: Group Runner 1.8 and MDMR Changes

### PR DESCRIPTION
**Pull request by diegoaper**
## Fixes

Fixes #1720 by @shnizzedy
Related to https://github.com/FCP-INDI/fcp-indi.github.io/issues/288 & https://github.com/FCP-INDI/C-PAC/issues/1214 by @sgiavasis & https://github.com/FCP-INDI/C-PAC/issues/1721 by @diegoaper 

## Description
- Fixed MDMR script bugs found in the translation from R to Python.
- Scripts: mdmr.py, cwas.py, pipeline.py, cpac_group_runner.py, group_config_template.yml
- Updating the group runner to follow the individual template.
- Allows group runner to take CPAC 1.8 BIDS directory for group analysis.

## Tests
Conducted tests between the local R scripts from the original scripts to the CPAC pipeline. As well used 2 independent datasets to see if the MDMR calculations were similar. This resulted in very strong correlation (r = 0.997) for both datasets and formulas.

## Screenshots
Allegra_dataset
Prisma_dataset

## Checklist

- [X] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [X] My pull request targets the `develop` branch of the repository. 
- [X] My commit messages follow best practices.
- [X] My code follows the established code style of the repository.
- [X] I added tests for the changes I made (if applicable).
- [ ] I updated the changelog.
- [X] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: CPAC/pipeline/cpac_group_runner.py [hunk 1]
```diff
@@ -176,25 +174,23 @@ def gather_nifti_globs(pipeline_output_folder, resource_list,
...
-
+        glob_string = os.path.join(pipeline_output_folder, "*", "*", 
+                                   "*"+resource_name+"*")
+        print(glob_string)
         # get all glob strings that result in a list of paths where every path
         # ends with a NIFTI file
-
         prog_string = ".."
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:186]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 2]
```diff
@@ -381,7 +376,7 @@ def create_output_dict_list(nifti_globs, pipeline_output_folder,
     '''
 
     # grab MeanFD_Jenkinson just in case
-    search_dirs += ["power_params"]
+    search_dirs += ["framewise-displacement-jenkinson"]
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:379]: Good catch

File: CPAC/pipeline/cpac_group_runner.py [hunk 3]
```diff
@@ -392,23 +387,23 @@ def create_output_dict_list(nifti_globs, pipeline_output_folder,
         for filename in files:
 
             filepath = os.path.join(root, filename)
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:390]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 4]
```diff
@@ -392,23 +387,23 @@ def create_output_dict_list(nifti_globs, pipeline_output_folder,
...
             filepath = os.path.join(root, filename)
-
+            
             if not any(fnmatch.fnmatch(filepath, pattern) for pattern in nifti_globs):
                 continue
 
             if not any(filepath.endswith(ext) for ext in exts):
                 continue
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:396]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 5]
```diff
@@ -392,23 +387,23 @@ def create_output_dict_list(nifti_globs, pipeline_output_folder,
...
-
+            
             relative_filepath = filepath.split(pipeline_output_folder)[1]
             filepath_pieces = [_f for _f in relative_filepath.split("/") if _f]
-
-            resource_id = filepath_pieces[1]
-
+            
+            resource_id = '_'.join(filepath_pieces[2].split(".")[0].split("_")[3:])
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:401]:
```suggestion

            resource_id = '_'.join(filepath_pieces[2].split(".")[0].split("_")[3:])

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 6]
```diff
@@ -525,14 +520,14 @@ def pheno_sessions_to_repeated_measures(pheno_df, sessions_list):
     More info:
       https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FEAT/
           UserGuide#Paired_Two-Group_Difference_.28Two-Sample_Paired_T-Test.29
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:523]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 7]
```diff
@@ -525,14 +520,14 @@ def pheno_sessions_to_repeated_measures(pheno_df, sessions_list):
...
-
+    
     Sample input:
         pheno_df
           sub01
           sub02
         sessions_list
           [ses01, ses02]
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:530]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 8]
```diff
@@ -41,18 +40,19 @@ def load_config_yml(config_file, individual=False):
...
 
     if individual:
-        config.pipeline_setup['log_directory']['path'] = os.path.abspath(config.pipeline_setup['log_directory']['path'])
-        config.pipeline_setup['working_directory']['path'] = os.path.abspath(config.pipeline_setup['working_directory']['path'])
-        config.pipeline_setup['output_directory']['path'] = os.path.abspath(config.pipeline_setup['output_directory']['path'])
-        config.pipeline_setup['crash_log_directory']['path'] = os.path.abspath(config.pipeline_setup['crash_log_directory']['path'])
+        config.logDirectory = os.path.abspath(config.logDirectory)
+        config.workingDirectory = os.path.abspath(config.workingDirectory)
+        config.outputDirectory = os.path.abspath(config.outputDirectory)
+        config.crashLogDirectory = os.path.abspath(config.crashLogDirectory)
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:55]: Are these supposed to be different than like https://github.com/FCP-INDI/C-PAC/blob/6dc4a2b904529504a21a91252d4dbfa3b4418fc8/CPAC/pipeline/cpac_group_runner.py#L1176-L1177 https://github.com/FCP-INDI/C-PAC/blob/6dc4a2b904529504a21a91252d4dbfa3b4418fc8/CPAC/pipeline/cpac_group_runner.py#L1337-L1340 ?

File: CPAC/pipeline/cpac_group_runner.py [hunk 9]
```diff
@@ -1293,7 +1290,7 @@ def run_cwas_group(pipeline_dir, out_dir, working_dir, crash_dir, roi_file,
                 df_dct[strat_scan] = strat_df[strat_df["Series"] == strat_scan]
         else:
             df_dct[list(set(strat_df["Series"]))[0]] = strat_df
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1293]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 10]
```diff
@@ -1325,37 +1328,46 @@ def run_cwas(pipeline_config):
...
                    inclusion=inclusion)
 
 
 def find_other_res_template(template_path, new_resolution):
-    """Find the same template/standard file in another resolution, if it
+    """
+    Find the same template/standard file in another resolution, if it
     exists.
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1364]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 11]
```diff
@@ -1325,37 +1328,46 @@ def run_cwas(pipeline_config):
...
+    """
+    Find the same template/standard file in another resolution, if it
     exists.
-
+    
     template_path: file path to the template NIfTI file
+    
     new_resolution: (int) the resolution of the template file you need
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1368]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 12]
```diff
@@ -1474,28 +1486,29 @@ def launch_PyBASC(pybasc_config):
...
 def run_basc(pipeline_config):
-    """Run the PyBASC module.
-
+    """
+    Run the PyBASC module.
+    
     PyBASC is a separate Python package built and maintained by Aki Nikolaidis
     which implements the BASC analysis via Python.
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1494]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 13]
```diff
@@ -1474,28 +1486,29 @@ def launch_PyBASC(pybasc_config):
...
+    
     PyBASC is based off of the following work:
         - Garcia-Garcia, M., Nikolaidis, A., Bellec, P., Craddock, R. C., Cheung, B., Castellanos, F. X., & Milham, M. P. (2017).
               Detecting stable individual differences in the functional organization of the human basal ganglia. NeuroImage.
         - Bellec, P., Rosa-Neto, P., Lyttelton, O. C., Benali, H., & Evans, A. C. (2010).
               Multi-level bootstrap analysis of stable clusters in resting-state fMRI. Neuroimage, 51(3), 1126-1139.
         - Bellec, P., Marrelec, G., & Benali, H. (2008).
               A bootstrap test to investigate changes in brain connectivity for functional MRI. Statistica Sinica, 1253-1268.
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1502]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 14]
```diff
@@ -1474,28 +1486,29 @@ def launch_PyBASC(pybasc_config):
...
         - Bellec, P., Rosa-Neto, P., Lyttelton, O. C., Benali, H., & Evans, A. C. (2010).
               Multi-level bootstrap analysis of stable clusters in resting-state fMRI. Neuroimage, 51(3), 1126-1139.
         - Bellec, P., Marrelec, G., & Benali, H. (2008).
               A bootstrap test to investigate changes in brain connectivity for functional MRI. Statistica Sinica, 1253-1268.
-
+    
     PyBASC GitHub repository:
         https://github.com/AkiNikolaidis/PyBASC
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1505]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 15]
```diff
@@ -1474,28 +1486,29 @@ def launch_PyBASC(pybasc_config):
...
-
+    
     PyBASC GitHub repository:
         https://github.com/AkiNikolaidis/PyBASC
-
+    
     PyBASC author:
         https://www.researchgate.net/profile/Aki_Nikolaidis
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1508]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 16]
```diff
@@ -1474,28 +1486,29 @@ def launch_PyBASC(pybasc_config):
...
-
+    
     PyBASC author:
         https://www.researchgate.net/profile/Aki_Nikolaidis
-
+    
     Inputs
         pipeline_config: path to C-PAC pipeline configuration YAML file
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1511]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 17]
```diff
@@ -1508,7 +1521,7 @@ def run_basc(pipeline_config):
            selected to run PyBASC for (preprocessed and template-space
            functional time series are pulled from each pipeline output
            directory, for input into PyBASC).
-        6. Gather functional_to_standard outputs from each pipeline.
+        6. Gather space-template_bold outputs from each pipeline.
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1524]: :sunglasses:

File: CPAC/pipeline/cpac_group_runner.py [hunk 18]
```diff
@@ -1518,7 +1531,7 @@ def run_basc(pipeline_config):
            into a config YAML file for each pipeline-strategy-scan we are
            running.
         10. Launch PyBASC for each configuration generated.
-
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1534]:
```suggestion

```

File: CPAC/pipeline/cpac_group_runner.py [hunk 19]
```diff
@@ -1610,7 +1624,8 @@ def run_basc(pipeline_config):
         roi_file_two = resample_cpac_output_image(roi_two_cmd_args)
         basc_config_dct['cross_cluster_mask_file'] = roi_file_two
 
-    pipeline_dir = os.path.abspath(pipeconfig_dct["pipeline_dir"])
+    pipeline_dir = os.path.abspath(pipeconfig_dct["pipeline_setup"]
+    ["output_directory"]["source_outputs_path"])
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1628]:
```suggestion
    pipeline_dir = os.path.abspath(pipeconfig_dct["pipeline_setup"]
                                   ["output_directory"]["source_outputs_path"])
```

File: CPAC/pipeline/cpac_group_runner.py [hunk 20]
```diff
@@ -1746,26 +1760,26 @@ def run_isc_group(pipeline_dir, out_dir, working_dir, crash_dir,
...
+        derivatives=["space-template_bold", "desc-Mean_timeseries"],
+        #exts=['nii', 'nii.gz', 'csv']
     )
 
     iteration_ids = []
-    for preproc_strat in output_df_dct.keys():
+    for preproc_strat in output_df_dct.items():
         # go over each preprocessing strategy
 
         derivative, _ = preproc_strat
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:1774]: I assume this is working, but I wonder why this change would be necessary. Like https://github.com/FCP-INDI/C-PAC/blob/6dc4a2b904529504a21a91252d4dbfa3b4418fc8/CPAC/pipeline/cpac_group_runner.py#L1794 uses `preproc_strat` as a key. Is that key a tuple like you'd get out of `output_df_dct.items()`?

File: CPAC/pipeline/cpac_group_runner.py [hunk 21]
```diff
@@ -2069,25 +2083,26 @@ def run(config_file):
...
     # Run randomise, if selected
-    if 1 in c.run_randomise:
+    if 1 in c["fsl_randomise"]["run"]:
         run_feat(config_file, feat=False)
 
     #Run QPP, if selected
-    if 1 in c.run_qpp:
+    if 1 in c["qpp"]["run"]:
         run_qpp(config_file)
+
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_group_runner.py:2108]:
```suggestion

```

File: CPAC/pipeline/engine.py [hunk 22]
```diff
@@ -1949,9 +1949,6 @@ def initiate_rpool(wf, cfg, data_paths=None, part_id=None):
                 ingress_raw_func_data(wf, rpool, cfg, data_paths, unique_id,
                                       part_id, ses_id)
 
-    # grab already-processed data from the output directory
-    rpool = ingress_output_dir(cfg, rpool, unique_id, creds_path)
-
```
Comment (shnizzedy) [inline CPAC/pipeline/engine.py:1954]: :raised_eyebrow: We don't want outputs from previous runs?

File: CPAC/resources/configs/group_config_template.yml [hunk 23]
```diff
@@ -1,340 +1,334 @@
...
 
 # General Group-Level Analysis Settings
-##############################################################################
 
-# The main input of group-level analysis- the output directory of your individual-level analysis pipeline run (pre-processing & derivatives for each participant). This should be a path to your C-PAC individual-level run's pipeline folder, which includes the sub-directories labeled with the participant IDs.
-pipeline_dir: /path/to/output_dir
+pipeline_setup:
 
+  # Name for this pipeline configuration - useful for identification.
+  pipeline_name: cpac-blank-template
```
Comment (shnizzedy) [inline CPAC/resources/configs/group_config_template.yml:14]:
```suggestion
  pipeline_name: cpac-group-template
```

**Review by shnizzedy — CHANGES_REQUESTED** [review]
This is great :clap:!

Most of my suggestions are just lint/style things, but there are a few real questions too.

Also: do [the user docs](https://fcp-indi.github.io/docs/nightly/user/group_mdmr) need any updates regarding these changes?

File: CPAC/resources/configs/group_config_template.yml [hunk 24]
```diff
@@ -1,340 +1,334 @@
 # CPAC Group-Level Analysis Configuration YAML file
-# Version 1.8.4.dev
+# Version 1.8.3
```
[the remark this answers is not in the document]
Comment (shnizzedy) [reply CPAC/resources/configs/group_config_template.yml]: I see you fixed this already after I started my review :smile_cat: https://github.com/FCP-INDI/C-PAC/pull/1787/commits/b6ed208a307dde12be98df6a1b3f8b39163d5b1a

File: CPAC/pipeline/cpac_group_runner.py (hunk 20 above)
Comment (diegoaper) [reply CPAC/pipeline/cpac_group_runner.py:1774]: I think so, its from the original code and didn't appear to be an error when I ran it.

File: CPAC/cwas/pipeline.py [hunk 25]
```diff
@@ -1,7 +1,8 @@
 
 import os
-from CPAC.pipeline import nipype_pipeline_engine as pe
+import nipype.pipeline.engine as pe
```
[the remark this answers is not in the document]
Comment (diegoaper) [reply CPAC/cwas/pipeline.py:3]: That's fine, I think it was from the previous 1.7.2 engine.

File: CPAC/cwas/pipeline.py [hunk 26]
```diff
@@ -95,20 +99,24 @@ def create_cwas(name='cwas', working_dir=None, crash_dir=None):
     workflow = pe.Workflow(name=name)
     workflow.base_dir = working_dir
     workflow.config['execution'] = {'hash_method': 'timestamp',
-                                    'crashdump_dir': os.path.abspath(crash_dir)}
+                                    'crashdump_dir': os.path.abspath(crash_dir), 
+                                    'crashfile_format': 'txt'}
```
[the remark this answers is not in the document]
Comment (diegoaper) [reply CPAC/cwas/pipeline.py:102]: Thank you!

File: CPAC/cwas/mdmr.py [hunk 27]
```diff
@@ -83,15 +83,15 @@ def mdmr(D, X, columns, permutations):
         Gs[:, di] = gower(D[di]).flatten()
     
     X1 = np.hstack((np.ones((subjects, 1)), X))
-    columns = columns.copy() + 1
+    columns = columns.copy() #removed a +1
```
[the remark this answers is not in the document]
Comment (diegoaper) [reply CPAC/cwas/mdmr.py:86]: Yes it was. I think it was how the Phenotypic CSV file was parsed in Pandas. It added an extra column for the subject list and then it ingress it without the SubID. It ended up crashing the runs until I removed the +1 to the columns. It was offsetting the variable of interest and messing with the distance calculation.

File: CPAC/pipeline/cpac_group_runner.py (hunk 8 above)
Comment (diegoaper) [reply CPAC/pipeline/cpac_group_runner.py:55]: Yes they are, completely missed them. Thank you!

Comment (diegoaper) [reply CPAC/pipeline/cpac_group_runner.py:55]: Looking back at this, I don't see how it would work since it would just replace the same object in the config dictionary.

File: CPAC/pipeline/cpac_group_runner.py (hunk 20 above)
Comment (diegoaper) [reply CPAC/pipeline/cpac_group_runner.py:1757]: I see, there are 2 other instances where this appears and it seems to do the same thing. However, I think .keys() is correct instead of .items().

File: CPAC/pipeline/engine.py (hunk 22 above)
Comment (diegoaper) [reply CPAC/pipeline/engine.py:1954]: What do you mean?

Comment (shnizzedy) [reply CPAC/pipeline/engine.py:1954]: I guess this one is a question for @sgiavasis 

In `develop` we had
https://github.com/FCP-INDI/C-PAC/blob/a4ee983c5bd5e8c2adabc54daefff5871a455099/CPAC/pipeline/engine.py#L1942-L1954
, but in this branch we have https://github.com/FCP-INDI/C-PAC/blob/6dc4a2b904529504a21a91252d4dbfa3b4418fc8/CPAC/pipeline/engine.py#L1944-L1953 (removing the lines this comment thread is on https://github.com/FCP-INDI/C-PAC/blob/a4ee983c5bd5e8c2adabc54daefff5871a455099/CPAC/pipeline/engine.py#L1950-L1951)

The change comes from this commit: https://github.com/diegoaper/C-PAC_MDMR-Group_Runner/commit/94d80829eee156ead693da1bbacb1f7e2dca3fd0 ; I'm just asking if this change is intended

File: CPAC/pipeline/cpac_group_runner.py (hunk 8 above)
Comment (shnizzedy) [reply CPAC/pipeline/cpac_group_runner.py]: I'm asking about what looks like https://github.com/FCP-INDI/C-PAC/blob/6dc4a2b904529504a21a91252d4dbfa3b4418fc8/CPAC/pipeline/cpac_group_runner.py#L52-L55 would take this structure
```YAML
logDirectory:
workingDirectory:
outputDirectory:
crashLogDirectory:
```
and update the paths to absolute paths, but other sections like https://github.com/FCP-INDI/C-PAC/blob/6dc4a2b904529504a21a91252d4dbfa3b4418fc8/CPAC/pipeline/cpac_group_runner.py#L1176-L1177 https://github.com/FCP-INDI/C-PAC/blob/6dc4a2b904529504a21a91252d4dbfa3b4418fc8/CPAC/pipeline/cpac_group_runner.py#L1337-L1340 look like they'd use a structure like
```YAML
pipeline_setup:
  log_directory:
    path:
  output_directory:
    source_outputs_path:
    output_path:
  working_directory:
    path:
```

Comment (diegoaper) [reply CPAC/pipeline/cpac_group_runner.py]: Ah I see what you mean, yes I'll change them for the config tree.

File: CPAC/pipeline/cpac_group_runner.py (hunk 20 above)
Comment (shnizzedy) [reply CPAC/pipeline/cpac_group_runner.py]: Resolved in https://github.com/FCP-INDI/C-PAC/pull/1787/commits/92f9658a869432f86e7e2cbfcfb8f4a9a8675f9c

**Review by shnizzedy — APPROVED** [review]
This PR looks good to me, and https://github.com/FCP-INDI/C-PAC/pull/1787#discussion_r990502579 was introduced in a separate PR (https://github.com/FCP-INDI/C-PAC/pull/1764), so we don't need to fix it to merge this PR

**Comment by shnizzedy** [comment]
I approved the changes, but I do still [have this question](https://github.com/FCP-INDI/C-PAC/pull/1787#pullrequestreview-1134597653):

> Also: do [the user docs](https://fcp-indi.github.io/docs/nightly/user/group_mdmr) need any updates regarding these changes?

Also, if you want to [update](https://github.com/FCP-INDI/.github/blob/main/CONTRIBUTING.md#changelog) the changelog (in the top "unreleased" section https://github.com/FCP-INDI/C-PAC/blob/68e34164d1392e457a2c181ea64fbf1e39cdc644/CHANGELOG.md?plain=1#L17-L32), that'll save us from needing to do it in a separate PR

**Comment by shnizzedy** [comment]
> I approved the changes, but I do still [have this question](https://github.com/FCP-INDI/C-PAC/pull/1787#pullrequestreview-1134597653):
> 
> > Also: do [the user docs](https://fcp-indi.github.io/docs/nightly/user/group_mdmr) need any updates regarding these changes?
> 

― https://github.com/FCP-INDI/C-PAC/pull/1787#issuecomment-1273799222

Update documented in https://github.com/FCP-INDI/fcp-indi.github.io/pull/292

File: CPAC/pipeline/engine.py (hunk 22 above)
Comment (sgiavasis) [reply CPAC/pipeline/engine.py:1954]: I wonder if this change got pulled in from elsewhere by accident, because _I_ wanted to remove that functionality for now (it has no real effect on the pipeline but was a stub for future expansion of this feature- plus fmriprep output ingress is far too not-seamless (seamful?) for it to be useful right now). And I already made a similar change in another branch, but based off `develop`.

Either way, it can stay - some of my current `develop`-based branches are doing the same.

**Comment by sgiavasis** [comment]
Awesome, nice @diegoaper !!
